src/inmanta/moduletool.py

```diff
@@ -181,7 +183,7 @@ def determine_new_version(self, old_version, version, major, minor, patch, dev):
             if not dev:
                 outversion = baseversion
             else:
-                outversion = "%s.dev%d" % (baseversion, time.time())
+                outversion = cast(Version, parse_version("%s.dev%d" % (baseversion, time.time())))
```

Comment:
Actual change request: I don't think we should cast this. This is not the same class (although they may have the same implementation). We should get rid of `pkg_resources` for iso8 anyway so this will be resolved then.

Reply:
is the vendored version an actual string?

Reply:
It's a `Version`-like object, just not of the type `packaging.Version`.

Reply:
but then we need to at least change something, for example use parse_version everywhere. because now we are mixing strings and version objects

Reply:
My change request was a bit ambiguous. I did not mean to say we should not make the return type uniform. We just shouldn't pretend like `parse_version` returns a `Version` and all is well, so I don't like the `cast` here.

When I wrote this I didn't realize this lived in relatively new code. In this instance I think it's actually safe to just use `packaging.version.Version` directly. Then we get rid of the deprecated method and the type errors both.

Reply:
you kind of lost me, can you make the changes you propose?

Reply:
Done

Reply:
```diff
diff --git a/src/inmanta/moduletool.py b/src/inmanta/moduletool.py
index ed831e5c..2fb3bac2 100644
--- a/src/inmanta/moduletool.py
+++ b/src/inmanta/moduletool.py
@@ -183,7 +183,7 @@ class ModuleLikeTool:
             if not dev:
                 outversion = baseversion
             else:
-                outversion = cast(Version, parse_version("%s.dev%d" % (baseversion, time.time())))
+                outversion = Version("%s.dev%d" % (baseversion, time.time()))
         else:
             opts = [x for x in [major, minor, patch] if x]
             if version is not None:
@@ -207,9 +207,9 @@ class ModuleLikeTool:
                     outversion = str(VersionOperation.set_version_tag(old_version, version_tag=""))

             if dev:
-                outversion = cast(Version, parse_version("%s.dev%d" % (outversion, time.time())))
+                outversion = Version("%s.dev%d" % (outversion, time.time()))

-        outversion = cast(Version, parse_version(outversion))
+        outversion = Version(outversion)
         if outversion <= old_version:
             LOGGER.error(f"new versions ({outversion}) is not larger then old version ({old_version}), aborting")
             return None
```

Reply:
I would then keep the part where you make this method consistent by not returning strings, but I'd not use the `cast` so that we don't cover up the lie in the return type.

Reply:
I think I fixed it by making sure we never use the pkg_resource parse_version anymore